frame/system/src/extensions/check_weight.rs

```diff
@@ -157,12 +157,18 @@ impl<T: Trait + Send + Sync> CheckWeight<T> where
 	}
 
 	/// get the priority of an extrinsic denoted by `info`.
+	///
+	/// Operational transaction will be given a fixed initial amount to be fairly distinguished from
+	/// the normal ones.
 	fn get_priority(info: &DispatchInfoOf<T::Call>) -> TransactionPriority {
 		match info.class {
-			DispatchClass::Normal => info.weight.into(),
-			// Don't use up the whole priority space, to allow things like `tip`
-			// to be taken into account as well.
-			DispatchClass::Operational => TransactionPriority::max_value() / 2,
+			// Normal transaction.
+			DispatchClass::Normal =>
+				FrameTransactionPriority::Normal(info.weight.into()).into(),
```

Comment:
why do we use weight as priority here ?

isn't it what the issue is about:

> This SignedExtension is bumping the priority by weight
Which is sort of dumb, why would higher weight transaction have higher priority thinking

it seems more correct to me to return 0, but I might miss some consideration